PowerShell_Scripts/FolderSynchronisation_edited+func_2ed.ps1

```diff
@@ -0,0 +1,27 @@
+﻿Clear-Host
+
+Write-Host "We will ask you to indicate the path for folder 1 and folder 2"
+function Path($TestFolderPath) {
+    Write-Host "Please, specify the path"
+    $TestFolderPath = Read-Host
+
+    return $TestFolderPath
+}
+
+$TestFolder1Path = Path #'C:\Users\Daria\Desktop\TestFolder1'
+$TestFolder2Path = Path #'C:\Users\Daria\Desktop\TestFolder2'
+
+$Folder1Content = Get-ChildItem $TestFolder1Path -Recurse
+$Folder2Content = Get-ChildItem $TestFolder2Path -Recurse
+
+$Difference1 = Compare-Object -ReferenceObject $Folder1Content -DifferenceObject $Folder2Content `
+-Property Name, Length -PassThru | Where-Object {$_.SideIndicator -eq "<="}
+$Difference1 | ForEach-Object {
+    Copy-Item $Folder1Content -Destination $TestFolder2Path -Force
+}
+
+$Difference2 = Compare-Object -ReferenceObject $Folder2Content -DifferenceObject $Folder1Content `
+-Property Name, Length -PassThru | Where-Object {$_.SideIndicator -eq "<="}
+$Difference2 | ForEach-Object {
+    Copy-Item $Folder2Content -Destination $TestFolder1Path -Force
+}
```

Comment:
This block of code is practically identical to the one above it - it differs only in variables, not the logic. This is a perfect opportunity to utilize a function: the logic will be written only once and will stay in one place, we'll just need to call the function twice with different arguments.

Reply:
Sure, I've made it this way:
```suggestion
function CompareContent{
    param($folderpath, $foldercontent1, $foldercontent2)
    $Difference = Compare-Object -ReferenceObject $foldercontent1 -DifferenceObject $foldercontent2 `
    -Property Name, Length -PassThru | Where-Object {$_.SideIndicator -eq "<="}
    $Difference | ForEach-Object {
    Copy-Item $foldercontent1 -Destination $folderpath -Force
    }
}

CompareContent $TestFolder2Path $Folder1Content $Folder2Content
CompareContent $TestFolder1Path $Folder2Content $Folder1Content
```

Reply:
That's way better. 👍  Still, there are some ways for improvement:
* Instead of passing the actual content of folders, pass folder paths only and get the content inside the function;
* The function says that it compares content of two folders but it actually _changes_ it. This is an semantic oversight you should always strive to avoid. The best option to resolve this is to separate the process into two functions: `Compare-Content` and `Copy-DifferentContent`, first of which will do the comparison and the other will do the copying;